hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestRecordLevelIndexWithSQL.scala

```diff
@@ -70,45 +77,45 @@ class TestRecordLevelIndexWithSQL extends RecordLevelIndexTestBase {
     assertEquals(5, spark.sql("select * from " + sqlTempTable).count())
 
     // non existing entries in EqualTo query
-    var dataFilter: Expression = EqualTo(attribute("_row_key"), Literal("xyz"))
+    var dataFilter: Expression = EqualTo(attribute(colName), Literal("xyz"))
     assertEquals(0, spark.sql("select * from " + sqlTempTable + " where " + dataFilter.sql).count())
     assertEquals(0, fileIndex.listFiles(Seq.empty, Seq(dataFilter)).flatMap(s => s.files).size)
 
     // non existing entries in IN query
-    dataFilter = In(attribute("_row_key"), List.apply(Literal("xyz"), Literal("abc")))
+    dataFilter = In(attribute(colName), List.apply(Literal("xyz"), Literal("abc")))
```

Comment:
How do we make sure the record level index is used in pruning in the test, i.e., will the test still pass if record level index is absent or not used?

Reply:
The test only enables RLI (apart from the file index by default). In `verifyPruningFileCount` we assert the file count with both data skipping enabled and disabled. In the case when skipping is enabled, we assert the filtered file count obtained from file index is strictly less than that when skipping is disabled. This could not have happened if RLI was not used.

Reply:
Sg